src/NetTopologySuite/Geometries/ILineal.cs

```diff
@@ -11,4 +11,45 @@ public interface ILineal
     {
         
     }
-}
\ No newline at end of file
+
+    /// <summary>
+    /// Marker interface to identify all <c>Geometry</c> subclasses that have a <c>Dimension</c> of
+    /// <see cref="Dimension.Curve"/> and consist of <b>only</b> <c>1</c> component.
+    /// </summary>
+    public interface ICurve : ILineal
```

Comment:
This interface is necessary because e.g. 
* `CompoundCurve` can consist of `LineString` and `CircularString` objects, 
* `CurvePolygon` can consist of  `LineString`, `CircularString` and `CompoundCurve` objects forming a ring, and 
* `MultiCurve` can consist of can consist of  `LineString`, `CircularString` and `CompoundCurve`

`LineString` can't be a base class to lineal geometries. `Curve` and `Surface` are present in OGC SFS Common Architecture.

Reply:
Could we not then add `Curve` and `Surface` as (abstract) base classes for `LineString` and `Polygon`, respectively?  It's a bit of a breaking change, but it seems preferable to adding public interfaces that don't really **do** anything.

Reply:
We'd have to have `LineString` derive from `Curve` and `Polygon` from `Surface`. That`d be a breaking change.

Reply:
I was trying to avoid that, but if you think that is better, I can live with that just as well.

Reply:
I keep losing this link: https://docs.microsoft.com/en-us/dotnet/core/compatibility lists the standards that the .NET team lives up to.  Introducing a new base class (like what I proposed as `Curve` and `Surface`) is actually listed as "requires judgment", since there are ways to do it that **would** break compat.

I think what I'm proposing should actually be fine (reflection breaking change at most, which doesn't matter for us since the new library version doesn't have the same compat requirements as .NET Framework libraries that need to work as in-place replacements for the old versions).